test/server/test_fxa_auth.py

```diff
@@ -0,0 +1,78 @@
+import urlparse
+import urllib
+from clientlib import ScreenshotsClient
+from urlparse import urljoin
+import random
+import requests
+import time
+from requests import HTTPError
+
+# Hack to make this predictable:
+random.seed(0)
+
+
+def get_url(url, expect=None):
+    result = urllib.urlopen(url)
+    if expect and result.getcode() != expect:
+        raise Exception("GET {} returned status {}".format(url, result.getcode()))
+    return result.read()
+
+# test we can change expiration and delete shots made previously
+def test_owned_shot_with_fxa_auth():
+    first_user = ScreenshotsClient()
+    second_user = ScreenshotsClient()
+
+    first_user.login();
+    shot_url = first_user.create_shot(docTitle="A_TEST_SITE_1")
+    shot_id = urlparse.urlsplit(shot_url).path.strip("/")
+    shot_page = first_user.read_shot(shot_url)
+    print(first_user.read_shot(shot_url)["clip_url"], shot_page["clip_url"])
+    assert first_user.read_shot(shot_url)["clip_content"] == shot_page["clip_content"]
+    assert "A_TEST_SITE_1" in shot_page["page"]
+
+
+    # Use first_user accountId to login to a second session
+    second_user.fxa_login(first_user.accountId)
```

Comment:
I guess because this never calls `second_user.login()`, therefore this gets an accountId cookie but no deviceId, but it's not obvious. Maybe it would be fine if after instantiation we just set `second_user.device_id = None` to be double-sure it's not half-logged-in.

From this perspective, maybe it should be something like `second_user.browser_only_login(accountId)`, and that method could ensure that the cookies were cleared, deviceId was unset, and the appropriate accountId cookie was added to the session.

Reply:
With browser only user created as instance of ScreenshotsBase class we don't have deviceId or accountId assigned at initialization.  Should we check for any previous existing accountId cookie?